Create new CS for multifactor authentication (MFA)

rbsec opened the pull request:
Draft PR to track progress on the new MFA cheat sheet (as per issue #96). This is **not ready to be merged**.

Any comments or feedback are welcome, but please comment in there or in the issue if you want to write/rewrite/add sections before doing so, otherwise we might both end up writing the same thing.

rbsec commented:
The initial draft of this is ready for comments and review. At this stage I'm less interested in grammar/stylistic changes (although these are important before it gets merged into main cheat sheets), and more in the content. Specifically:

* Is it factually correct?
* Is it useful?
* Is there important stuff missing?
* Is there stuff that should be removed?

ThunderSon commented on cheatsheets_draft/Multifactor_Authentication_Cheat_Sheet.md:
What would be the other mean, other than a cookie?

rbsec replied:
I was thinking a cookie + IP address pair. Updated in bf6a9e0422abfc818fcf91f0ed7b13d38cefdecd.

ThunderSon commented on cheatsheets_draft/Multifactor_Authentication_Cheat_Sheet.md:
Some websites use the email to login users. So no passwords, just emails sent to authenticate a user. I think this can fit in here not only as a 2FA, but as a password replacement.

rbsec replied:
I've added a note about this in bf6a9e0422abfc818fcf91f0ed7b13d38cefdecd - is that the sort of thing you were thinking?

rbsec replied:
@ThunderSon are you happy with this change?

mackowski commented:
Just and update from my side, I started reviewing it and it looks very good. I will finish review over this weekend.

mackowski commented on cheatsheets_draft/Multifactor_Authentication_Cheat_Sheet.md:
broken link it should be `#software-totp-tokens`

rbsec replied:
Fixed in e917e0b

rbsec commented:
@mackowski thanks for the review - I've fixed that broken link and added some comments about whether the various methods are resistant to phishing.

Most were fairly obvious, but I'm in two minds about biometrics. It might be possible in some cases to phish them (for example, taking a photo that you would then use to present to a badly made photo recognition system), but well designed biometrics should be pretty hard to phishing (cloning handprints, etc).

I'm thinking it's best to just not talk about phishing for biometrics/location - thoughts?

mackowski commented on cheatsheets_draft/Multifactor_Authentication_Cheat_Sheet.md:
this point should be in `Pros` section

rbsec replied:
Fixed in 580272e.

mackowski commented:
> I'm thinking it's best to just not talk about phishing for biometrics/location - thoughts?

I agree, this depends on technology/implementation. If we are not sure I agree that it's best to not talk about it in this case.

mackowski commented:
@rbsec looks good for me. Do you want to improve it further or do you want to merge it?

rbsec commented:
@mackowski I don't think that we'll get much more feedback until it's pushed live and a few more people see it.

I wonder if we should have a new cheat sheet tag or something like that in the page/intro, so people to try and encourage a bit more feedback and so people view is as being slightly immature?
